oauthproxy.go

```diff
@@ -445,9 +446,10 @@ func buildRoutesAllowlist(opts *options.Options) ([]allowedRoute, error) {
 		var (
 			method string
 			path   string
+			negate = strings.Contains(methodPath, "!=")
 		)
 
-		parts := strings.SplitN(methodPath, "=", 2)
+		parts := regexp.MustCompile("!?=").Split(methodPath, 2)
```

Comment:
Why does the `strings.SplitN` no longer work?

Reply:
Because I must split on a regex: either `=` or `!=`